### DISCUSSION TITLE
Tokens sent to wrong mint

### DISCUSSION THREAD

**Discussion by yamijuan** [wormhole-foundation/wormhole#314] [Problems / bugs] [answered]
Hi. I'm working with Solana people trying to figure out some of the issues presented in discord. The latest ones being with renLuna tokens. Looks like the site tried to send back WWT (renluna on eth) to solana to the wrong mint associated account but correct owner. is there a fix/workaround? so users can get their tokens back? One example is this tx
https://etherscan.io/tx/0x9be08755f4740ace5eed90eac2e0beb5e5d74ed72b5ba3b9d64d11e710d4de6b

the tokens were tried to send to this account: https://explorer.solana.com/address/JBsAgjLFUwjyBEVUMcWpjrYKmKNDEbqcXazSrfENXTuu

but as you can see that's the wrong token mint

**Comment by leoluk** [comment]
How do you know it's the wrong mint?

The token account is owned by the user, so they should be able to revert the transfer.

**Comment by yamijuan** [comment]
well the token bridge-d is renLuna but the recipient associated account is `unknown token` the funds never got there I believe or how can they look at the transfer and revert it? and revert it to where exactly?

**Reply by leoluk** [reply]
Tokens don't have names on Solana - are you sure that the `unknown token` isn't in fact wrappedRenLUNA?

**Reply by yamijuan** [reply]
it might be a possibility but I don't see any funds of that token in their wallet. do you think you can help me figure that out?  The problem was that `wrappedAssetMintKey` return that mintkey instead of the original renLuna one

**Comment by leoluk** [comment, marked as answer]
This was determined to be a UI bug in wormholebridge.com where the wrong mint was used to create the token account:

- https://github.com/solana-labs/oyster/issues/179
- https://github.com/solana-labs/oyster/issues/170
- https://github.com/solana-labs/oyster/issues/159

**Comment by leoluk** [comment]
First of all, I should note that this bug is not caused by Wormhole, but a third party UI (wormholebridge.com) completely unrelated to the Wormhole project, just like Metamask is a third party UI for Ethereum.

The good news is that the transfer is, as far as I can tell, likely provably invalid – there's no way to ever complete it, since it's impossible to create the correct token account (this needs to be double checked!).

This would mean that is is safe to revert the transfer by manually signing a VAA that returns the tokens on Ethereum.

This decision is entirely up to the Wormhole v1 governance - what do you think?

**Reply by tyrion70** [reply]
We (Chainlayer) support reverting the transaction by manually signing a VAA, assuming its indeed verified and impossible to create the correct token account.

**Reply by clawmvp** [reply]
We [01node] support reverting the tx.

**Reply by 1xCL** [reply]
We {Staking Fund} support the reverting.

**Reply by mcf-rocks** [reply]
Support revert

**Reply by Chainode** [reply]
Chainode Tech supports reverting the transaction if is proved to be a regular use, without any intentional malicious act behind.

**Reply by vshvsh** [reply]
I'm okay with doing this one (or several) times but wouldn't want to have to make these decisions on a permanent basis. We're ill-equipped to investigate every case and be a judge in it. 

P2P will ack this given the final confirmation.
